### DISCUSSION TITLE
Get data form ui with cgi

### DISCUSSION THREAD

**Discussion by Wiphat101** [AxisCommunications/acap-native-sdk-examples#277] [Q&A] [answered]
Hi, everyone
I want to get values ​​from Ui in the format [{\"x\":758,\"y\":128},{\"x\":772,\"y\":491}] using Cgi. What are the steps?
"acap-sdk version" : 3.5
"schemaVersion" : 1.3
thank

**Comment by Cacsjep** [comment]
Could you please explain it more.

**Reply by Wiphat101** [reply]
I want to receive Polygons values from drawing frames from Ui by sending the values to store in cgi.
The values obtained from polygons look like this: [{"x":758,"y":128},{"x":772,"y":491}]

**Reply by Cacsjep** [reply]
So you had an web frontend (UI) hosted via acap, and now you want store an array of positions via cgi on camera, that you want to send via an ajax js call from your web frontend? Is this correct or wrong?

**Reply by Wiphat101** [reply]
Yes, correct

**Reply by Cacsjep** [reply]
did you see this ? https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/web-server-using-fastcgi

**Reply by Wiphat101** [reply]
I have seen this but do you have a way to use it with the opc-ua-colorchecker-acap example? Because I am using CGI data transmission from this example.
https://github.com/AxisCommunications/opc-ua-colorchecker-acap

**Reply by Cacsjep** [reply, marked as answer]
OK, did you See this ? https://github.com/AxisCommunications/opc-ua-colorchecker-acap/blob/2dab70f13a359b5072275413fbd4700629c8c77a/src/opcuacolorchecker.cpp#L497 i think you need to add an new cgi path to your manifest and then add in the request handler where you check against the new cgi path like here: https://github.com/AxisCommunications/opc-ua-colorchecker-acap/blob/2dab70f13a359b5072275413fbd4700629c8c77a/src/opcuacolorchecker.cpp#L514

**Reply by Cacsjep** [reply]
Here in manifest you need to add the new cgi path: https://github.com/AxisCommunications/opc-ua-colorchecker-acap/blob/2dab70f13a359b5072275413fbd4700629c8c77a/manifest.json#L21

**Reply by Wiphat101** [reply]
If I want to get the polygon drawing position from the front end using an array variable, what should I do? Because of this value There will be at least 4 sets: [x1,y1... x4,y4]

**Reply by pataxis** [reply]
Just mentioning that it should be possible to convert the color checker example to use ACAP Native SDK and FastCGI.
In ACAP Native SDK you also have a supported JSON library jansson that could perhaps be used by the CGI for polygon data.
It's a bit of work to do it of course.
